### PR TITLE
fix: escape regex characters in string searches(mongo)

### LINKED ISSUE #1266: Need to escape special characters in string matching filters
For example the following filter for mongo:
```
{
  "filter": {
    "title": {
      "contains": "y..t"
    }
      }
}

```
returns 
```
{
  "data": {
    "findNotes": {
      "items": [
        {
          "title": "123123y..t213",
          "description": "a"
        },
        {
          "title": "123123yeet213",
          "description": "a"
        }
      ]
    }
  }
}
```
 Also the following filter for postgres:
```
{
  "filter": {
    "title": {
      "contains": "y__t"
    }
      }
}

```
returns
```
{
  "data": {
    "findNotes": {
      "items": [
        {
          "title": "123123yeet213",
          "description": "a"
        },
        {
          "title": "123123y__t213",
          "description": "a"
        }
      ]
    }
  }
}
```

Should we escape the characters or add a new feature to the docs :P?

### PR DESCRIPTION
**Pull request by ssd71**
Fixes #1266 for mongo

# Verification

```
mutation create {
  createNote(input: { title: "23yeet39", description: "1" }) {
    title,
    id,
    description
  }
}

query find {
  findNotes(filter: { title: { contains: "y..t" } }) {
    items {
      title
    }
  }
}
```

### CODE REVIEW

**Comment by wtrocki** [comment]
I will be back to this sometime today.

File: packages/graphback-runtime-mongodb/package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "@graphback/runtime": "0.14.0-dev2",
     "@types/mongodb": "3.5.18",
     "dataloader": "2.0.0",
+    "escape-string-regexp": "^4.0.0",
```
Comment (craicoverflow) [inline packages/graphback-runtime-mongodb/package.json:42]: Remove ^

**Comment by wtrocki** [comment]
Tried tested - Very happy :D

**Comment by machi1990** [comment]
A bit late, but I'd suggest adding an integration test with this change.

**Comment by wtrocki** [comment]
Never too late ;)

**Pull request merged**